pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -630,17 +631,24 @@ class Utils {
                 }
             }
 
-            // Remove the triggers/parameters properties regardless.
-            j.removeProperty(PipelineTriggersJobProperty.class)
+            def currentTriggerProp = j.getProperty(PipelineTriggersJobProperty.class)
+            def newTriggerProp = new PipelineTriggersJobProperty(triggersToApply)
+            // Only Remove/Add if the triggers are different.
+            if (!areTriggersInPipelineTriggerJobPropertiesEqual(currentTriggerProp, newTriggerProp)) {
+                j.removeProperty(PipelineTriggersJobProperty.class)
+                if (!triggersToApply.isEmpty()) {
+                    j.addProperty(newTriggerProp)
```

Comment:
```suggestion
                    j.addProperty(newTriggersProp)
```

Also, this replaces all the triggers even if only one has changed, right?
It would be better if made it so only the _changed_ triggers are replaced.  But that might be work for another PR.